packages/editor/src/store/actions.js

```diff
@@ -599,6 +599,20 @@ export function removeNotice( id ) {
 	};
 }
 
+/**
+ * Returns an action object used to lock the editor.
+ *
+ * @param {Object}  lock Details about the post lock status, user, and nonce.
+ *
+ * @return {Object} Action object.
+ */
+export function updatePostLock( lock ) {
+	return {
+		type: 'UPDATE_POST_LOCK',
+		lock,
+	};
+}
```

Comment:
Action creators are also useful as contracts on the data expected to be flowing through the system, and can be either manually consulted by the developer or relied upon via text editors' completion features. In this case, `lock` is a bit opaque, whereas we could have something like:

```js
updatePostLock( lockStatus, user, … ) {
  return {
    type: 'UPDATE_POST_LOCK',
    lockStatus,
    user,
    …
  };
}
```

alternatively:

```js
return {
  type: 'UPDATE_POST_LOCK',
  lock: { lockStatus, user, … },
};
```

I don't mind keeping the action creator as it is now, but this was worth pointing out.